test/integration/certificates/issuing_controller_test.go

```diff
@@ -200,7 +204,7 @@ func TestIssuingController(t *testing.T) {
 	apiutil.SetCertificateCondition(crt, crt.Generation, cmapi.CertificateConditionIssuing, cmmeta.ConditionTrue, "", "")
 	crt.Status.NextPrivateKeySecretName = &nextPrivateKeySecretName
 	crt.Status.Revision = &revision
-	crt, err = cmCl.CertmanagerV1().Certificates(namespace).UpdateStatus(ctx, crt, metav1.UpdateOptions{})
+	err = internalcertificates.ApplyStatus(ctx, cmCl, fieldManager, crt)
```

Comment:
> And patching the status will update the Metadata.ResourceVersion right? Or is that totally irrelevant when using server-side-apply?

Yes as I understand the resource version is irrelevant. I think that the resource version being different is what makes our Update operations fail on conflicts (as the version is different in etcd because it has been bumped due to other controller's changes etc). With SSA resource version can be omitted and I _think_ that happens when we [deepcopy object meta](https://github.com/cert-manager/cert-manager/blob/master/internal/controller/certificates/apply.go#L78) as [deepcopy function omits the resource version field](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/zz_generated.deepcopy.go#L662-L708).

I _think_ that 'apply status' functions don't return a resource simply because it's never needed after a status update..
In tests where I was making changes I mostly needed the returned object only to have the correct `metadata.generation`  field which is used by [`WaitForCertificateReadyAndDoneIssuing`](https://github.com/cert-manager/cert-manager/blob/9411a4c7d99adea3339b445f771391ef5c8f078e/test/e2e/framework/helper/certificates.go#L95) function.

Reply:
> With SSA resource version can be omitted and I think that happens when we [deepcopy object meta](https://github.com/cert-manager/cert-manager/blob/master/internal/controller/certificates/apply.go#L78) as [deepcopy function omits the resource version field](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/zz_generated.deepcopy.go#L662-L708).

(In general I think it would be good if we had functionality that can extract all fields owned by a field manager and only include those in a spec update like it's done in [client-go applyconfigurations](https://github.com/kubernetes/client-go/tree/master/applyconfigurations) so that we don't accidentally co-own other manager's field when applying a spec update. Perhaps the code generators that we use should be able to generate such functionality (as well as `Apply` methods) for custom resources.
But I think this is not an issue for now as we don't really have different managers managing object specs.

Reply:
It looks like there is an effort towards that in:
 * https://github.com/kubernetes-sigs/controller-runtime/pull/1365/files#diff-2ff41033efc095081a9f65e1551a189610980864776c4ebf2d72c9a0865b18e1R4
 * https://github.com/kubernetes-sigs/controller-tools/pull/536

Reply:
Thanks for those links, I wasn't aware of that work!
So, we need to keep an eye on this

Reply:
Ahh, I see that there already is https://github.com/kubernetes/code-generator/tree/master/cmd/applyconfiguration-gen to generate applyconfigurations. I will see if this can be used to generate applyconfigurations for our custom types.
I guess then it would be just the ability to generate Apply method for our typed clients that would be missing.